### PR TITLE
[docs] Update Functions package content

### PR DESCRIPTION
**Pull request by Jennifer88huang**
### Motivation
On [Functions package page](https://pulsar.apache.org/docs/en/next/functions-package/),  the same notes/tips occur many times, and some code snippets are not displayed correctly.

### Modifications
1. Re-org the structure, reduce note/tips
2. Fix code snippet display issues.
3. Fix indent issues.

### CODE REVIEW

**Comment by Jennifer88huang** [comment]
@sijia-w @Huanli-Meng please help review the content, any comment is highly appreciated.

File: site2/docs/functions-package.md [hunk 1]
```diff
@@ -4,58 +4,24 @@ title: Package Pulsar Functions
...
-This section provides step-by-step instructions to package Pulsar functions in Java, Python, and Go. 
+You can package Pulsar functions in Java, Python, and Go. Packaging a window function in Java is the same as [packaging a function in Java](#java).
 
-> **Tip**
->
-> - Packaging a window function in Java is the same as [packaging a function in Java](#java) as below. 
->
-> - Currently, the window function is not available in Python and Go.
+> **Note**    
+> Currently, window function is not available in Python and Go.
```
Comment (Huanli-Meng) [inline site2/docs/functions-package.md:10]:
```suggestion
> Currently, the window function is unavailable in Python and Go.
```

File: site2/docs/functions-package.md [hunk 2]
```diff
@@ -4,58 +4,24 @@ title: Package Pulsar Functions
 sidebar_label: "How-to: Package"
 ---
 
-This section provides step-by-step instructions to package Pulsar functions in Java, Python, and Go. 
+You can package Pulsar functions in Java, Python, and Go. Packaging a window function in Java is the same as [packaging a function in Java](#java).
```
Comment (Huanli-Meng) [inline site2/docs/functions-package.md:7]: a window function or the window function?

File: site2/docs/functions-package.md [hunk 3]
```diff
@@ -4,58 +4,24 @@ title: Package Pulsar Functions
...
-> **Tip**
->
-> - `$PWD/data` is the local directory. `-v` maps the `/pulsar/data` directory in the Docker image to the local `$PWD/data` directory.
->
-> - To check whether the image starts, use the command `docker ps`.
-
-### Run Pulsar cluster in k8s
-
-For details about how to deploy a Pulsar cluster in the k8s environment, For details, see [here](helm-overview.md).
+Before running a Pulsar function, you need to start Pulsar. You can [run a standalone Pulsar in Docker](getting-started-docker.md), or [run Pulsar cluster in Kubernetes](getting-started-helm.md).
```
Comment (Huanli-Meng) [inline site2/docs/functions-package.md:14]: run pulsar cluster or run pulsar?

File: site2/docs/functions-package.md [hunk 4]
```diff
@@ -4,58 +4,24 @@ title: Package Pulsar Functions
...
-This example demonstrates how to package a function in Java.
-
-> **Note**
->
-> This example assumes that you have [run a standalone Pulsar in Docker](#run-a-standalone-pulsar-in-docker) successfully.
-
+To package a function in Java, complete the following steps.
 
-1. Create a new maven project with a pom file.
+1. Create a new maven project with a pom file. In the following code sample, `mainClass` is your package name.
```
Comment (Huanli-Meng) [inline site2/docs/functions-package.md:22]: it should be Maven

File: site2/docs/functions-package.md [hunk 5]
```diff
@@ -127,49 +93,41 @@ This example demonstrates how to package a function in Java.
         }
     }
     ```
+ 
+     For the package imported, you can use one of the following interfaces:
```
Comment (Huanli-Meng) [inline site2/docs/functions-package.md:97]:
```suggestion
     For the imported package, you can use one of the following interfaces:
```

File: site2/docs/functions-package.md [hunk 6]
```diff
@@ -127,49 +93,41 @@ This example demonstrates how to package a function in Java.
...
+     ```
 
 3. Package the Java function.
 
     ```bash
     mvn package
     ```
 
-    After the Java function is packaged, a `target` directory is automatically created. Open the `target` directory to see if there is a jar package similar to `java-function-1.0-SNAPSHOT.jar`.
+    After the Java function is packaged, a `target` directory is created automatically. Open the `target` directory to see if there is a jar package similar to `java-function-1.0-SNAPSHOT.jar`.
```
Comment (Huanli-Meng) [inline site2/docs/functions-package.md:130]:
```suggestion
    After the Java function is packaged, a `target` directory is created automatically. Open the `target` directory to check if there is a JAR package similar to `java-function-1.0-SNAPSHOT.jar`.
```

File: site2/docs/functions-package.md [hunk 7]
```diff
@@ -289,45 +233,31 @@ This example demonstrates how to package a function by **one python file** in Py
...
-    > - If you want to start a function in cluster mode, replace `localrun` with `create` in the command above. The following log indicates that the Python function starts successfully.
-    >
-    >   ```text
-    >   "Created successfully"
-    >   ```
-
 ### ZIP file
 
-This example demonstrates how to package a function by **ZIP file** in Python.
+To package a function with **ZIP file** in Python, complete the following steps.
```
Comment (Huanli-Meng) [inline site2/docs/functions-package.md:238]:
```suggestion
To package a function with the **ZIP file** in Python, complete the following steps.
```

File: site2/docs/functions-package.md [hunk 8]
```diff
@@ -449,34 +360,30 @@ This example demonstrates how to package a function in Go.
...
+    ```
+    if fc, ok := pf.FromContext(ctx); ok {
+        fmt.Printf("function ID is:%s, ", fc.GetFuncID())
+        fmt.Printf("function version is:%s\n", fc.GetFuncVersion())
+     }
+     ```
+
+    When writing a Go function, remember that
+    - In `main()`, you **only** need to register the function name to `Start()`. **Only** one function name is received in `Start()`. 
+    - Go function uses Go reflection based on the received function name to verify whether the parameter list and returned value list implemented are correct. The parameter list and returned value list specified **must be** one of the following sample functions:
```
Comment (Huanli-Meng) [inline site2/docs/functions-package.md:374]: I think this sentence needs to be refined:
whether the parameter list and returned value list implemented are correct.
The parameter list and returned value list specified **must be** one of the following sample functions:

**Review by Huanli-Meng — COMMENTED** [review]
Leave my comments, PTAL

File: site2/docs/functions-package.md (hunk 4 above)
Comment (Jennifer88huang) [reply site2/docs/functions-package.md:22]: @wolfstudy please confirm, thank you.

File: site2/docs/functions-package.md (hunk 8 above)
Comment (Jennifer88huang) [reply site2/docs/functions-package.md:374]: nice catch, I can simplify it, thanks for your reminder.

**Comment by Jennifer88huang** [comment]
@Huanli-Meng thanks for your comments. PTAL again.

**Review by wolfstudy — APPROVED** [review]
LGTM +1

**Pull request merged**